### PR TITLE
feat: add rule no-str-format

### PR DESCRIPTION
**Pull request by reka**

### CODE REVIEW

File: use-fstring.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# https://docs.sourcery.ai/Reference/Default-Rules/refactorings/use-fstring-for-formatting/
+
+rules:
+
+# You can use this rule in the Sourcery CLI with the --enable option:
+#
+# sourcery review --enable no-str-format .
+#
+- id: no-str-format
+  description: Use fstring instead of str.format
```
Comment (Hellebore) [inline use-fstring.yaml:17]: I would like an explanation here - short paragraph on why f-strings are better

File: use-fstring.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# https://docs.sourcery.ai/Reference/Default-Rules/refactorings/use-fstring-for-concatenation/
+# https://docs.sourcery.ai/Reference/Default-Rules/refactorings/use-fstring-for-formatting/
+
+rules:
+
+# You can use this rule in the Sourcery CLI with the --enable option:
+#
+# sourcery review --enable no-str-format .
+#
+- id: no-str-format
```
Comment (Hellebore) [inline use-fstring.yaml:16]: slighlty worried about name-space clashes now that we have rules coming in from various different places - not entirely sure what to do about it.

File: use-fstring.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  description: Use fstring instead of str.format
+  pattern: ${text}.format(...)
+  condition: text.has_type("str")
```
Comment (Hellebore) [inline use-fstring.yaml:19]: I think these rules need some basic positive and negative tests.

**Review by Hellebore — CHANGES_REQUESTED** [review]
It's a nice intro rule - It needs an explanation and some tests.

File: use-fstring.yaml (hunk 2 above)
Comment (reka) [reply use-fstring.yaml:16]: That's an excellent point. :+1: 

I've created a test PR #3 , and it seems our CI currently doesn't catch ID clashes.

Suggestion:

* Add a test to the internal CI that ensures that each ID is unique among the (default U optional) rules
* As long as this test doesn't exist: We should verify during the review that there's no ID clash. (I guess a text search for the ID in the internal repo is sufficient.)

Comment (reka) [reply use-fstring.yaml:16]: The name `no-str-format` seems to be available. :heavy_check_mark:

Comment (Hellebore) [reply use-fstring.yaml:16]: Could you add such.a test? Would be better than the manual check.

Comment (reka) [reply use-fstring.yaml:16]: I've created an internal issue for it.

Comment (Hellebore) [reply use-fstring.yaml:16]: Please add the test rather than waiting for the internal issue - should be fairly straightforward.

**Review by Hellebore — CHANGES_REQUESTED** [review]

**Pull request closed without merging**